packages/playground/define/vite.config.js

```diff
@@ -15,6 +15,7 @@ module.exports = {
         }
       }
     },
-    'process.env.SOMEVAR': '"SOMEVAR"'
+    'process.env.SOMEVAR': '"SOMEVAR"',
+    'process.env.SOMEOBJ': { spread: true }
```

Comment:
Why would `'process.env.SOMEOBJ': { spread: true }` every be a possible value?
I think `process.env.SOMETHING` will ever be a string, something other than a string should not be allowed or be parse with `JSON.parse`

Reply:
True. I hadn't really thought about that. It's a silly example. 

I'll update it to be make more sense, such as: `{ ...process.env.NODE_ENV === 'dev' ? { dev: true } : {} }`

Reply:
Lets say `process.env.NODE_ENV` is defined as `'dev'`
Even then `...process.env.NODE_ENV === 'dev'` would be `false` cause e.g. `...'dev'` would spread the string into something like a char array :eyes: 
So why and when would you need spread for a process env variable?

Reply:
It's not spreading the env variable, but rather using it in a ternary operator that spreads another object. I was using it in a config object where I needed to add conditional config based on an env variable.

```js
const config = {
  ...process.env.NODE_ENV === 'dev' ? { watch: true } : { minify: true }
}
```

I know that code could be written a lot of different ways, and I've since refactored it, but I thought I'd send in a PR since it took me some debugging and I figured someone else might run into the same issue at some point.

The same issue would appear when spreading something from env into an array, which might be more common than my use case. Something like `const sequence = [ ...process.env.SOME_NUMBERS ]`.

Reply:
Wait... is 

```js
const config = {
  ...process.env.NODE_ENV === 'dev' ? { watch: true } : { minify: true }
}
```

even allowed JS?
Shouldn't it be at least e.g. 

```js
const config = {
  ...(process.env.NODE_ENV === 'dev' ? { watch: true } : { minify: true })
}
```

Reply:
> Wait... is
> 
> ```js
> const config = {
>   ...process.env.NODE_ENV === 'dev' ? { watch: true } : { minify: true }
> }
> ```
> 
> even allowed JS?
> Shouldn't it be at least e.g.
> 
> ```js
> const config = {
>   ...(process.env.NODE_ENV === 'dev' ? { watch: true } : { minify: true })
> }
> ```

I got confused by that at first as well. But it's allowed.

<img width="383" alt="Screen Shot 2021-07-27 at 10 04 58" src="https://user-images.githubusercontent.com/4687577/127121063-72ea8262-db03-4e4e-a988-b1b60b8f3c0e.png">

> And also `const sequence = [ ...process.env.SOME_NUMBERS ]` should already be allowed, isn't it?
> I would think it will result in e.g. `['0', '1', '2']` where `process.env.SOME_NUMBERS` equals to `'012'`

It will still fail the RegExp because of the preceding `...`. I updated the tests to include that as well.

> So are you sure this PR is really fixing anything? And if so, what?
> And then please update the tests that should fail without the changes.
> And tests should not contain invalid usages of `process.env`

I have cleaned the tests and verified that this fix does work. Both tests I added will fail without the addition to the RegExp:

<img width="712" alt="Screen Shot 2021-07-27 at 10 15 39" src="https://user-images.githubusercontent.com/4687577/127121289-90e76f42-980d-45a8-8926-799681689b00.png">
<img width="684" alt="Screen Shot 2021-07-27 at 10 16 03" src="https://user-images.githubusercontent.com/4687577/127121295-d8cb7213-6e3f-4415-b7fb-1612400a948b.png">

After the RegExp fix:

<img width="453" alt="Screen Shot 2021-07-27 at 10 16 24" src="https://user-images.githubusercontent.com/4687577/127121325-179b63bb-595b-4f60-8810-0154f7a3ff58.png">